### ISSUE TITLE
sign in using facebook oauth

### ISSUE THREAD

**Issue by gka** [gka/schnack#18]

**Comment by jhabdas** [comment]
Banned in China. Would encourage avoiding SSO with Internet gatekeepers like Facebook and Google, and, instead, considering going with something like [Civic](https://www.civic.com/) for decentralized identity management longer-term.

**Comment by gka** [comment]
I agree about the gatekeeper part, but the reason this issue exists is that some users have brought it up. there's little use in a auth provider that nobody is using.

Personally I don't use Facebook and try not to use Google, but I can see that for other website owner this might be different. In the end it depends on where you think your audience is. For someone in Russia or China adding WeChat or VKontakte would make even more sense.

Not sure if anyone is still using OpenID? Wasn't that supposed to be the decentralized alternative?

Civic sounds great, never heard of it. But I didn't see it implemented in [Passport.js](http://www.passportjs.org/).

**Comment by jhabdas** [comment]
I'm onboard with letting users help guide the product direction. Oftentimes users at large are not aware of the gatekeeper and SSO issues social uses as a user retention method, much like Facebook and Instagram do not allow photo exports to ensure you don't ever leave (I learned this the hard way before publishing [4 Years Without Facebook](https://medium.com/@jhabdas/what-i-learned-in-4-years-without-facebook-981fb68a64e2) (may currently be restricted by Medium for some users).

Here's the passport implementation for Civic: https://github.com/jaredhanson/passport/issues/597. I tried their iOS app and it's solid for what it needs to do. I wager their timing was dead on the money to make them the de factor future standard for identity and, later, the place where the real money is—reputation systems (for better or worse).

Multiple providers is handy. Having a pluggable abstraction layer like Passport makes that possible.
